lib/blocks.php

```diff
@@ -9,64 +9,34 @@
 	die( 'Silence is golden.' );
 }
 
-$wp_registered_blocks = array();
-
 /**
- * Registers a block.
+ * Registers a block type.
  *
- * @param  string $name Block name including namespace.
- * @param  array  $settings Block settings.
-
- * @return array            The block, if it has been successfully registered.
+ * @since 0.1.0
+ *
+ * @param string $name Block type name including namespace.
+ * @param array  $args {
+ *     Array of block type arguments. Any arguments may be defined, however the following
+ *     ones are supported by default.
+ *
+ *     @type callable $render Callback used to render blocks of this block type.
+ * }
+ * @return WP_Block_Type|false The registered block type on success, or false on failure.
  */
-function register_block_type( $name, $settings ) {
-	global $wp_registered_blocks;
-
-	if ( ! is_string( $name ) ) {
-		$message = __( 'Block names must be strings.' );
-		_doing_it_wrong( __FUNCTION__, $message, '0.1.0' );
-		return false;
-	}
-
-	$name_matcher = '/^[a-z0-9-]+\/[a-z0-9-]+$/';
-	if ( ! preg_match( $name_matcher, $name ) ) {
-		$message = __( 'Block names must contain a namespace prefix. Example: my-plugin/my-custom-block' );
-		_doing_it_wrong( __FUNCTION__, $message, '0.1.0' );
-		return false;
-	}
-
-	if ( isset( $wp_registered_blocks[ $name ] ) ) {
-		/* translators: 1: block name */
-		$message = sprintf( __( 'Block "%s" is already registered.' ), $name );
-		_doing_it_wrong( __FUNCTION__, $message, '0.1.0' );
-		return false;
-	}
-
-	$settings['name'] = $name;
-	$wp_registered_blocks[ $name ] = $settings;
-
-	return $settings;
+function register_block_type( $name, $args ) {
+	return WP_Block_Type_Registry::get_instance()->register( $name, $args );
 }
 
 /**
- * Unregisters a block.
+ * Unregisters a block type.
  *
- * @param  string $name Block name.
- * @return array        The previous block value, if it has been
- *                      successfully unregistered; otherwise `null`.
+ * @since 0.1.0
+ *
+ * @param string $name Block type name including namespace.
+ * @return WP_Block_Type|false The unregistered block type on success, or false on failure.
```

Comment:
What benefit is there to returning a `WP_Block_Type` object just after it has been unregistered?
In case someone wants to amend it and re-register it?